pyat/at/lattice/elements/magnet_elements.py

```diff
@@ -436,12 +495,38 @@ def items(self) -> Generator[tuple[str, Any], None, None]:
 
 
 class Octupole(Multipole):
-    """Octupole element, with no changes from multipole at present"""
+    """Octupole element"""
 
-    _BUILD_ATTRIBUTES = Multipole._BUILD_ATTRIBUTES
+    _BUILD_ATTRIBUTES = LongElement._BUILD_ATTRIBUTES + ["O"]
+    _stacklevel = 7  # Stacklevel for warnings
 
     DefaultOrder = 3
 
+    def __init__(self, family_name: str, length: float, o: float = 0.0, **kwargs):
+        """
```

Comment:
Problem here: you change the signature of the constructor which was
```
Octupole(name, length, polya, poly)
```
which is not backward compatible. This is illustrated by the change you had to do in other files.

You could try 
```python
def __init__(self, name, length, *args, **kwargs):

    if isinstance(args[0], Iterable):
        polya, polyb = args[:2]
    else:
        polya = []
        polyb = [0.0, 0.0, 0.0, args[0]]

    kwargs.setdefault("PassMethod", "StrMPoleSymplectic4Pass")
    super().__init__(family_name, length, poly, poly, **kwargs)
```
to keep compatibility